src/test/java/org/kiwiproject/registry/eureka/util/EurekaTestDataHelper.java

```diff
@@ -0,0 +1,178 @@
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON;
+import static javax.ws.rs.core.MediaType.APPLICATION_JSON_TYPE;
+import static org.awaitility.Awaitility.await;
+import static org.kiwiproject.base.KiwiStrings.f;
+import static org.kiwiproject.jaxrs.KiwiResponses.successful;
+import static org.kiwiproject.registry.eureka.config.EurekaRegistrationConfig.DEFAULT_LEASE_EXPIRATION_DURATION_SECONDS;
+import static org.kiwiproject.registry.eureka.config.EurekaRegistrationConfig.DEFAULT_LEASE_RENEWAL_INTERVAL_SECONDS;
+import static org.kiwiproject.registry.eureka.server.EurekaRegistryService.DEFAULT_DATA_CENTER_INFO_CLASS;
+import static org.kiwiproject.registry.eureka.server.EurekaRegistryService.DEFAULT_DATA_CENTER_NAME;
+import static org.kiwiproject.registry.eureka.server.EurekaRegistryService.LEASE_DURATION_IN_SECONDS;
+import static org.kiwiproject.registry.eureka.server.EurekaRegistryService.LEASE_RENEWAL_INTERVAL_IN_SECONDS;
+import static org.kiwiproject.test.jaxrs.JaxrsTestHelper.assertNoContentResponse;
+import static org.kiwiproject.test.jaxrs.JaxrsTestHelper.assertNotFoundResponse;
+import static org.kiwiproject.test.jaxrs.JaxrsTestHelper.assertOkResponse;
+
+import lombok.experimental.UtilityClass;
+import lombok.extern.slf4j.Slf4j;
+import org.glassfish.jersey.message.GZipEncoder;
+import org.kiwiproject.jaxrs.KiwiGenericTypes;
+import org.kiwiproject.registry.eureka.common.EurekaInstance;
+import org.kiwiproject.registry.eureka.common.EurekaResponseParser;
+import org.kiwiproject.registry.model.ServiceInstance;
+import org.kiwiproject.registry.util.ServiceInfoHelper;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.client.ClientBuilder;
+import javax.ws.rs.core.Response;
+
+@UtilityClass
+@Slf4j
+public class EurekaTestDataHelper {
+
+    public static ImageFromDockerfile eurekaImage() {
+        return new ImageFromDockerfile()
+                .withFileFromClasspath("Dockerfile", "eureka-server/Dockerfile")
+                .withFileFromClasspath("eureka-server-1.10.17.war", "eureka-server/eureka-server-1.10.17.war")
+                .withFileFromClasspath("config.properties", "eureka-server/config.properties")
+                .withFileFromClasspath("eureka-client-test.properties", "eureka-server/eureka-client-test.properties")
+                .withFileFromClasspath("eureka-server-test.properties", "eureka-server/eureka-server-test.properties");
+    }
+
+    public static String eurekaUrl(GenericContainer container) {
+        var host = container.getHost();
+        var port = container.getFirstMappedPort();
+        return f("http://{}:{}/eureka/v2", host, port);
+    }
+
+    public static void waitForEurekaToStart(String baseUrl) {
+        var client = ClientBuilder.newClient().register(GZipEncoder.class);
```

Comment:
I think I've now seen this ( `register(GZipEncoder.class)` ) a few times. For this library, perhaps we create a factory method to create Jersey `Client` instances with the correct settings, which can be re-used when a Eureka `Client` is needed?